server/tests/src/test/java/org/infinispan/server/functional/RestOperations.java

```diff
@@ -73,6 +74,19 @@ public void testRestOperations() {
       assertEquals(protocol, response.getProtocol());
    }
 
+   @Test
+   public void testPutWithTimeToLive() throws InterruptedException {
+      RestClientConfigurationBuilder builder = new RestClientConfigurationBuilder();
+      builder.protocol(protocol);
+      RestClient client = SERVER_TEST.rest().withClientConfiguration(builder).create();
+      RestCacheClient cache = client.cache(SERVER_TEST.getMethodName());
+      sync(cache.post("k1", "v1", 1, 1));
+      assertEquals(HttpResponseStatus.OK.code(), sync(cache.get("k1")).getStatus());
+      Thread.sleep(2000);
+      assertEquals(HttpResponseStatus.NOT_FOUND.code(), sync(cache.get("k1")).getStatus());
```

Comment:
Again this seems fine, but it isn't related to the current commit and Jira ticket. If you want to add this test, create a new Jira and add as a separate commit please.